### PR TITLE
Clean caloParams

### PR DESCRIPTION
**Pull request by aloeliger**
### PR description:

This PR addresses https://github.com/cms-sw/cmssw/issues/42668

I have moved a large number of caloParams files that were cluttering the L1TCalorimeter directory into their own L1T managed repository here: https://github.com/cms-l1t-offline/caloParams. This repository can be cloned into `L1Trigger/L1TCalorimeter/python` and all imports should work as before. I have removed most of `L1Trigger/Configuration/python/customiseSettings` and have set it to try to search for the new caloParams directory, error-ing out with instructions if it is not found.

@bundocka and @mitaylor since you two work with the caloParams the most, I want to run this solution by you before it is committed.

### PR validation:

caloParams functions can be imported from the same path they were previously, provided this repository is present

### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

This PR is not a backport and is not expected to be backported.

### CODE REVIEW

**Comment by aloeliger** [comment]
please test

**Comment by bundocka** [comment]
Hi Andrew,
Thanks  a lot for this. Hopefully this will keep everyone happy. A few comments:

- I think we should keep the `caloParams_cfi` base class in the master, for clarity, and there may be other classes that rely on it
- We need to leave the cmsDriver customisations unaffected by this change, i.e. we should be able to run something like [1] and it be transparent to this PR. This prevents us receiving a flood of emails complaining about the existing workflows not working in latest CMSSW. 

Cheers,
Aaron

[1]
`cmsDriver.py l1Ntuple -s RAW2DIGI --python_filename=met.py -n 10 --era=Run3 --data --no_output --conditions=132X_dataRun3_Prompt_v4 --customise=L1Trigger/Configuration/customiseReEmul.L1TReEmulFromRAWCalo --customise=L1Trigger/Configuration/customiseSettings.L1TSettingsToCaloParams_2023_v0_3 --customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleAODRAWEMUCalo --filein=/store/data/Run2023C/Muon0/RAW-RECO/ZMu-PromptReco-v1/000/367/112/00000/e50aedd7-1801-4bb8-8456-c2f814b00958.root`

**Comment by smuzaffar** [comment]
Milestone for this pull request has been moved to CMSSW_14_0_X.Please open a backport if it should also go in to CMSSW_13_3_X.

**Comment by aloeliger** [comment]
Apologies all who have been waiting for this, this has been continually delayed, I am going to be taking a look at whether this is still a suitable solution now that the holidays are over.

**Comment by aloeliger** [comment]
@bundocka I had the chance to look

> Hi Andrew, Thanks a lot for this. Hopefully this will keep everyone happy. A few comments:
> 
> * I think we should keep the `caloParams_cfi` base class in the master, for clarity, and there may be other classes that rely on it
> * We need to leave the cmsDriver customisations unaffected by this change, i.e. we should be able to run something like [1] and it be transparent to this PR. This prevents us receiving a flood of emails complaining about the existing workflows not working in latest CMSSW.
> 
> Cheers, Aaron
> 
> [1] `cmsDriver.py l1Ntuple -s RAW2DIGI --python_filename=met.py -n 10 --era=Run3 --data --no_output --conditions=132X_dataRun3_Prompt_v4 --customise=L1Trigger/Configuration/customiseReEmul.L1TReEmulFromRAWCalo --customise=L1Trigger/Configuration/customiseSettings.L1TSettingsToCaloParams_2023_v0_3 --customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleAODRAWEMUCalo --filein=/store/data/Run2023C/Muon0/RAW-RECO/ZMu-PromptReco-v1/000/367/112/00000/e50aedd7-1801-4bb8-8456-c2f814b00958.root`

I can get this cmsDriver command to work with this using the customise settings, but you are right about the caloParams base class, otherwise every single existing calo params file would have to be modified to work, :disappointed:, let me try again with that change.

**Comment by aloeliger** [comment]
> Hi Andrew, Thanks a lot for this. Hopefully this will keep everyone happy. A few comments:
> 
> * I think we should keep the `caloParams_cfi` base class in the master, for clarity, and there may be other classes that rely on it
> * We need to leave the cmsDriver customisations unaffected by this change, i.e. we should be able to run something like [1] and it be transparent to this PR. This prevents us receiving a flood of emails complaining about the existing workflows not working in latest CMSSW.
> 
> Cheers, Aaron
> 
> [1] `cmsDriver.py l1Ntuple -s RAW2DIGI --python_filename=met.py -n 10 --era=Run3 --data --no_output --conditions=132X_dataRun3_Prompt_v4 --customise=L1Trigger/Configuration/customiseReEmul.L1TReEmulFromRAWCalo --customise=L1Trigger/Configuration/customiseSettings.L1TSettingsToCaloParams_2023_v0_3 --customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleAODRAWEMUCalo --filein=/store/data/Run2023C/Muon0/RAW-RECO/ZMu-PromptReco-v1/000/367/112/00000/e50aedd7-1801-4bb8-8456-c2f814b00958.root`

I have added the `caloParams` file back into CMSSW, and this cmsDriver command works up until the point where some datafile (a `.txt`) seems to be missing, but it finds and works the customizations well enough. I had to modify some strange file in CMSSW to make this all work though. 

@bundocka I had to modfy `hackConditions_cff.py` to make this all work, it seems to be some file looking for 2017 conditions for PP and HI runs. Do you know what this is?

**Comment by bundocka** [comment]
> @bundocka I had to modfy `hackConditions_cff.py` to make this all work, it seems to be some file looking for 2017 conditions for PP and HI runs. Do you know what this is?

hackConditions was used to manually load the conditions to configure the emulator before we could load them from the DB. This was deprecated some time ago, I think here https://github.com/cms-sw/cmssw/pull/20871

When you say "to make this all work", what exactly was failing? hackConditions shouldn't be called anymore anywhere (I can't see it anywhere in L1Trigger/ at least). In fact hackConditions should probably be removed.

**Comment by antoniovilela** [comment]
ping (to make bot change milestone)

**Comment by makortel** [comment]
Just checking, is this PR intended for Phase 2 (20_1_X) or Run 3 legacy (17_0_X) or both?
